server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```
@@ -398,7 +393,7 @@ private boolean assertSerialization(DocumentMapper mapper) {
         final CompressedXContent mappingSource = mapper.mappingSource();
         Mapping newMapping = parseMapping(mapper.type(), mappingSource);
         if (newMapping.toCompressedXContent().equals(mappingSource) == false) {
-            throw new IllegalStateException(
+            throw new AssertionError(
```

This was technically swallowing assertion failures so I fixed it.